src/compiler/compile/render_ssr/handlers/DynamicElement.ts

```diff
@@ -0,0 +1,214 @@
+import {
+	get_attribute_value,
+	get_class_attribute_value
+} from './shared/get_attribute_value';
+import { get_slot_scope } from './shared/get_slot_scope';
+import { boolean_attributes } from './shared/boolean_attributes';
+import Renderer, { RenderOptions } from '../Renderer';
+import DynamicElement from '../../nodes/DynamicElement';
+import ElementHandler from './Element';
+import { x } from 'code-red';
+import Expression from '../../nodes/shared/Expression';
+import remove_whitespace_children from './utils/remove_whitespace_children';
+import Element from '../../nodes/Element';
+
+export default function (
+	node: DynamicElement,
+	renderer: Renderer,
+	options: RenderOptions & {
+		slot_scopes: Map<any, any>;
+	}
+) {
+	const dependencies = node.tag.dynamic_dependencies();
+
+	if (dependencies.length === 0) {
+		((node as unknown) as Element).dynamic_tag = node.tag;
+		ElementHandler((node as unknown) as Element, renderer, options);
+	} else {
+		const children = remove_whitespace_children(node.children, node.next);
+
+		// awkward special case
+		let node_contents;
+
+		const contenteditable = node.attributes.some(
+			(attribute) => attribute.name === 'contenteditable'
+		);
+
+		const slot = node.get_static_attribute_value('slot');
+		const nearest_inline_component = node.find_nearest(/InlineComponent/);
+
+		if (slot && nearest_inline_component) {
+			renderer.push();
+		}
+
+		renderer.add_string('<');
+		renderer.add_expression(node.tag.node);
+
+		const class_expression_list = node.classes.map((class_directive) => {
+			const { expression, name } = class_directive;
+			const snippet = expression ? expression.node : x`#ctx.${name}`; // TODO is this right?
+			return x`${snippet} ? "${name}" : ""`;
+		});
+		// if (node.needs_manual_style_scoping) {
```

Comment:
we shouldn't be adding commented out code. can you remove or replace with a TODO or soemthing more explanatory?

Reply:
Thanks, I overlooked that part. Since this is a duplicate of the Element node, I've copied the `add_css_class` implementation from there.